File: Ejercicio 2/tablaMultiplicar.cpp
#include <iostream>
#include <sstream>
#include <string>
using namespace std;
 
int main() {
    int numero;
    int opcion;
    int i;
    bool numeroValido;
    string entrada;
 
    cout << "TABLA DE MULTIPLICAR SELECCIONADA" << endl;
 
    // Se valida el numero antes de mostrar el menu
    do {
        cout << "Ingrese un numero entre 1 y 12: ";
        getline(cin, entrada);
 
        stringstream conversor(entrada);
        char sobrante;
        numeroValido = (conversor >> numero) && !(conversor >> sobrante)
                       && numero >= 1 && numero <= 12;
 
        if (numeroValido) {
            cout << "Numero valido." << endl;
        } else {
            cout << "Numero invalido. Debe ser un entero entre 1 y 12." << endl;
        }
    } while (!numeroValido);
 
    // El menu se ejecuta al menos una vez y se repite hasta elegir Salir
    do {
        cout << "\n========== MENU ==========" << endl;
        cout << "1. Tabla ascendente" << endl;
        cout << "2. Tabla descendente" << endl;
        cout << "3. Salir" << endl;
        cout << "Elija una opcion: ";
        getline(cin, entrada);
 
        stringstream conversorOpcion(entrada);
        char sobrante;
        if (!(conversorOpcion >> opcion) || (conversorOpcion >> sobrante)) {
            opcion = 0;
        }
 
        switch (opcion) {
            case 1:
                cout << "\nTABLA ASCENDENTE DEL " << numero << endl;
                for (i = 1; i <= 12; i++) {
                    cout << numero << " x " << i << " = " << numero * i << endl;
                }
                break;
 
            case 2:
                cout << "\nTABLA DESCENDENTE DEL " << numero << endl;
                for (i = 12; i >= 1; i--) {
                    cout << numero << " x " << i << " = " << numero * i << endl;
                }
                break;
 
            case 3:
                cout << "Gracias por usar el programa. Hasta pronto." << endl;
                break;
 
            default:
                cout << "Opcion invalida. Elija 1, 2 o 3." << endl;
                break;
        }
    } while (opcion != 3);
 
    return 0;
}

File: Ejercicio 2/tablaMultiplicar.java
package ejercicio2;
import java.util.Scanner;
 
public class tablaMultiplicar {
    public static void main(String[] args) {
        int numero;
        int opcion;
        int i;
        boolean numeroValido;
 
        Scanner scanner = new Scanner(System.in);
 
        System.out.println("TABLA DE MULTIPLICAR SELECCIONADA");
 
        // Se valida el numero antes de mostrar el menu
        do {
            System.out.print("Ingrese un numero entre 1 y 12: ");
            String entradaNumero = scanner.nextLine();
 
            try {
                numero = Integer.parseInt(entradaNumero);
                numeroValido = numero >= 1 && numero <= 12;
 
                if (numeroValido) {
                    System.out.println("Numero valido.");
                } else {
                    System.out.println("Numero invalido. Debe estar entre 1 y 12.");
                }
            } catch (NumberFormatException e) {
                numero = 0;
                numeroValido = false;
                System.out.println("Entrada invalida. Ingrese un numero entero.");
            }
        } while (!numeroValido);
 
        // El menu se ejecuta al menos una vez y se repite hasta elegir Salir
        do {
            System.out.println("\n========== MENU ==========");
            System.out.println("1. Tabla ascendente");
            System.out.println("2. Tabla descendente");
            System.out.println("3. Salir");
            System.out.print("Elija una opcion: ");
 
            String entradaOpcion = scanner.nextLine();
            try {
                opcion = Integer.parseInt(entradaOpcion);
            } catch (NumberFormatException e) {
                opcion = 0;
            }
 
            switch (opcion) {
                case 1:
                    System.out.println("\nTABLA ASCENDENTE DEL " + numero);
                    for (i = 1; i <= 12; i++) {
                        System.out.println(numero + " x " + i + " = " + (numero * i));
                    }
                    break;
 
                case 2:
                    System.out.println("\nTABLA DESCENDENTE DEL " + numero);
                    for (i = 12; i >= 1; i--) {
                        System.out.println(numero + " x " + i + " = " + (numero * i));
                    }
                    break;
 
                case 3:
                    System.out.println("Gracias por usar el programa. Hasta pronto.");
                    break;
 
                default:
                    System.out.println("Opcion invalida. Elija 1, 2 o 3.");
                    break;
            }
        } while (opcion != 3);
 
        scanner.close();
    }
}

